test/unit/util.test.js

```diff
@@ -36,6 +36,39 @@ describe('UTILITY FUNCTION TESTS ', function () {
       done();
     });
 
+    it('should resolve circular structures', function(done) {
+      let schema = {
+          '$ref': '#/components/schemas/a'
+        },
+        components = {
+          schemas: {
+            'a': {
+              'type': 'array',
+              'items': {
+                '$ref': '#/components/schemas/b'
+              }
+            },
+            'b': {
+              'type': 'object',
+              'properties': {
+                'c': {
+                  '$ref': '#/components/schemas/a'
+                }
+              }
+            }
+          }
+        },
+        bodyType = 'REQUEST',
+
+        result = Utils.safeSchemaFaker(schema, bodyType, components),
+        tooManyLevelsString = result[0].c[0].c[0].c[0].c[0].c;
```

Comment:
@abhijitkane this thing we have to do if we have to check whether the circular refs have been resolved or not.